AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Migration/MutationSyncMetadataMigrationDelegate+SQLite.swift

```diff
@@ -0,0 +1,119 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import Foundation
+import SQLite
+import AWSPluginsCore
+
+final class SQLiteMutationSyncMetadataMigrationDelegate: MutationSyncMetadataMigrationDelegate {
+
+    let modelSchemas: [ModelSchema]
+    weak var storageAdapter: SQLiteStorageEngineAdapter?
+
+    init(storageAdapter: SQLiteStorageEngineAdapter, modelSchemas: [ModelSchema]) {
+        self.storageAdapter = storageAdapter
+        self.modelSchemas = modelSchemas
+    }
+
+    func transaction(_ basicClosure: BasicThrowableClosure) throws {
+        try storageAdapter?.transaction(basicClosure)
+    }
+
+    // MARK: - Clear
+
+    func clear() throws {
+        log.debug("Clearing MutationSyncMetadata and ModelSyncMetadata to force full sync.")
+        try deleteMutationSyncMetadata()
+        try deleteModelSyncMetadata()
+    }
+
+    @discardableResult func deleteMutationSyncMetadata() throws -> String {
+        guard let storageAdapter = storageAdapter else {
+            log.debug("Missing SQLiteStorageEngineAdapter")
+            throw DataStoreError.unknown("Missing storage adapter for model migration", "", nil)
+        }
+
+        return try storageAdapter.emptyStore(for: MutationSyncMetadata.schema)
+    }
+
+    @discardableResult func deleteModelSyncMetadata() throws -> String {
+        guard let storageAdapter = storageAdapter else {
+            log.debug("Missing SQLiteStorageEngineAdapter")
+            throw DataStoreError.unknown("Missing storage adapter for model migration", "", nil)
+        }
+
+        return try storageAdapter.emptyStore(for: ModelSyncMetadata.schema)
+    }
+
+    // MARK: - Migration
+
+    func migrate() throws {
+        log.debug("Modifying and backfilling MutationSyncMetadata")
+        try dropMutationSyncMetadataCopyIfExists()
+        try createMutationSyncMetadataCopyTable()
+        try backfillMutationSyncMetadata()
+        try dropMutationSyncMetadata()
+        try renameMutationSyncMetadataCopy()
+    }
+
+    @discardableResult func dropMutationSyncMetadataCopyIfExists() throws -> String {
+        guard let storageAdapter = storageAdapter else {
+            log.debug("Missing SQLiteStorageEngineAdapter")
+            throw DataStoreError.unknown("Missing storage adapter for model migration", "", nil)
+        }
+
+        return try storageAdapter.removeStore(for: MutationSyncMetadataCopy.schema)
+    }
+
+    @discardableResult func createMutationSyncMetadataCopyTable() throws -> String {
```

Comment:
if we decide to take the approach described in the above comment, some examples of storage agnostic names:
- `createMutationSyncMetadataCopyStorage`
- `createMutationSyncMetadataCopyCollection`

Reply:
over in the storage adapter, now conforms to
```
protocol StorageEngineMigrationAdapter {

    @discardableResult func removeStore(for modelSchema: ModelSchema) throws -> String

    @discardableResult func createStore(for modelSchema: ModelSchema) throws -> String

    @discardableResult func emptyStore(for modelSchema: ModelSchema) throws -> String

    @discardableResult func renameStore(from: ModelSchema, toModelSchema: ModelSchema) throws -> String
}
```

so will name these accordingly, like `createMutationSyncMetadataCopyStore`